src/codeu/chat/ServerMain.java

```diff
@@ -89,13 +89,13 @@ public static void main(String[] args) {
   private static void runServer(Uuid id,
                                 Secret secret,
                                 ConnectionSource serverSource,
-                                ConnectionSource relaySource) {
+                                ConnectionSource relaySource, File persistentPath) {
 
     final Relay relay = relaySource == null ?
                         new NoOpRelay() :
                         new RemoteRelay(relaySource);
 
-    final Server server = new Server(id, secret, relay);
+    final Server server = new Server(id, secret, relay, persistentPath);
```

Comment:
persistentPath is a directory, this is the source of the FileNotFoundException, which still happens when I start the Server.

You need to do something like `new File(persistentPath, 'somerandomfilename.txt')` instead of passing in the directory unchanged.